packages/react-components/src/AccessibilityScenarios/Input.stories.tsx

```diff
@@ -0,0 +1,209 @@
+
+  const handleControlChange = (event: React.FormEvent, data: InputOnChangeData, name: string) => {
+    const stateUpdate = { [name]: data.value };
+    setValues({ ...values, ...stateUpdate });
+  };
+  const handleShowPasswordChange = (event: React.ChangeEvent) => {
+    setIsPasswordVisible(!isPasswordVisible);
+  };
+  const handleControlBlur = (event: React.FormEvent, name: string) => {
+    validateControl(name);
+  };
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+    if (validateForm()) {
+      setIsSubmitted(true);
+    }
+  };
+
+  const validateForm = () => {
+    for (const name in validations) {
+      if (!validateControl(name)) {
+        // Set the focus to the invalid control
+        const validation = validations[name];
+        let invalidControl;
+        if (validation.type === 'field' || validation.type === 'password') {
+          invalidControl = document.getElementById(name);
+        }
+        invalidControl?.focus();
+        return false;
+      }
+    }
+    return true;
+  };
+
+  const validateControl = (name: string) => {
+    // First hide the error message if exists
+    const validation = validations[name];
+    let stateUpdate = { [name]: false };
+    setIsErrorMessageVisible({ ...isErrorMessageVisible, ...stateUpdate });
+
+    // If the control value is invalid, show the error and set the aria-invalid attribute
+    const value = values[name];
+    let isInvalid = false;
+    if (validation.type === 'field' || validation.type === 'password') {
+      for (let i = 0; i < validation.regexes.length; i++) {
+        if (!validation.regexes[i].exec(value)) {
+          isInvalid = true;
+          break;
+        }
+      }
+    }
+    if (isInvalid) {
+      // Set some timeout to give the element with role="alert" a chance to speak
+      setTimeout(() => {
+        stateUpdate = { [name]: true };
+        setIsErrorMessageVisible({ ...isErrorMessageVisible, ...stateUpdate });
+      }, 400);
+      stateUpdate = { [name]: isInvalid };
+      setInvalids({ ...invalids, ...stateUpdate });
+      return false;
+    }
+    return true;
+  };
+
+  return (
+    <Scenario pageTitle="Registration form inputs">
+      <h1>Registration form</h1>
+      {!isSubmitted ? (
+        <form onSubmit={handleSubmit}>
+          <Label htmlFor="fullName">Full name:</Label>
+          <Input
+            type="text"
+            id="fullName"
+            name="fullName"
+            onChange={(event, data) => {
+              handleControlChange(event, data, 'fullName');
+            }}
+            onBlur={(event: React.FocusEvent) => {
+              handleControlBlur(event, 'fullName');
+            }}
+            aria-required="true"
+            aria-describedby="fullNameHint fullNameHint1 fullNameHint2"
+          />
+          <ErrorMessage isVisible={isErrorMessageVisible.fullName}>
+            <p id="fullNameHint">Full name is invalid. It must</p>
```

Comment:
I think it would be more appropriate to put an ID on the error message wrapper, then use that as the aria-describedby of the Input, and remove the IDs for the individual li elements.

Reply:
True, I think I had some reason for having individual ids in place, but cannot remember it now, so I am changing the code as you suggested.